libs/platform/src/lib/components/menu/menu-component/menu-item.component.ts

```diff
@@ -69,7 +69,7 @@ export class MenuItemComponent implements DefaultMenuItem, OnChanges {
 
     /** @hidden */
     ngOnChanges(changes: SimpleChanges): void {
-        if (changes.itemWidth || changes.item) {
+        if (changes.itemWidth || changes.item || changes.selected) {
             this.finalItemWidth = this.getItemWidth();
```

Comment:
I think you may be able to remove the `getItemWidth` method and the `finalItemWidth` property using the CSS `calc()` method: https://developer.mozilla.org/en-US/docs/Web/CSS/calc

The `calc()` method is compatible with all browsers and allows you to add together mixed units.

You could probably change line 31 in the `menu-item.component.html` template file to:

```
<div class="fd-menu__label-truncation" [ngStyle]="{ width: 'calc(' + itemWidth + ' + 46px )' }">
```

Reply:
Okay, I will try this and update.

Reply:
I have been trying this, Kevin, and it is not as straightforward to do it in this particular case because I have to take care of three things: 
1. the width of the item when an icon is selected or not selected(offset by 46px)
2. the width of the item when a normal icon(not selectable) is used(offset by 46px)
3. the width of the secondary icon if it is used. 
If both primary and secondary icons are used, I need to offset by 88px, but if any one icon is used, I need to offset by 46px, and for no icons there is no offset. 
I tried using calc in css and using ngClass instead to give different offsets, but I cannot access the [previous sibling](https://stackoverflow.com/questions/1817792/is-there-a-previous-sibling-selector) so getting the 3rd case to work seamlessly is causing some problems.  There are other edge cases as well which I had worked out programmatically but doing it through CSS alone would be all the more complicated.

Reply:
OK. It sounds like your use case is complicated enough that it needs this added javascript code. Keep your original solution.